content/ADR-164-quests.poc.md

```diff
@@ -0,0 +1,211 @@
+
+- Kernel should be responsible of receiving and notifiyng updates to Explorer.
+- Kernel should be able to receive requests from Explorer to start a quest and ask for it to the System.
+- Kernel should be able to notify in world events to the System, in order to make progress.
+
+### Actions
+
+A set of predefined actions may be provided to the designers and users, in order to define the Quests steps and make progress. 
+
+- Location
+- NPC Interaction
+- Play Emote
+- TBD
+
+### Quest Designer
+
+Build a tool/CLI to easily design Quests, without conflicting with existing definitions.
+
+Using the predefined actions, the designer should be able to create a Quest definition with steps and requirements. The output of a quest definition is a graph which will be persisted in a convenient form.
+
+### Backend
+
+The Quests System will expose an API with convenient endpoints for Designers and Users. 
+
+It also will process all events received from Explorer/Kernel sessions and make progress to corresponding Quest instances.
+
+Users may ask for quests state and metrics, but completed Quests may be stored for a certain time if the system needs space or it may grow undefinetly.
+
+- The system should be able to process all events in parallel, in order to make progress in corresponding Quests.
+- The system should be able to accept connections and subscriptions to Quest instances.
+- The system should be able to send updates to subscribed users.
+- The system should be able to respond to Quests state requests.
+- The system should expose metrics related to the Quests.
+
+### Workflows: Quest User
+
+- Start Quest
+
+There would be several ways to discovery a Quest. The user may be interested to start any of them. 
+
+```mermaid
+sequenceDiagram
+    Quest Discovery-->>Explorer/Kernel: New quest available
+    Explorer/Kernel->>Quest Server: Send start quest request 
+    Quest Server ->> Explorer/Kernel: Send start quest response
+    loop
+        Quest Server -->> Explorer/Kernel: Send quest updates
+    end
+```
+
+- Track Progress 
+
+Client should make progress by sending verified events
+```mermaid
+sequenceDiagram
+    Explorer/Kernel ->> Catalyst: Ask for signed event 
+    Explorer/Kernel ->> Quest Server: Send signed event
+    loop
+        Quest Server -->> Explorer/Kernel: Send quest updates
+    end
+```
+
+- Real time updates
+
+Once the API is defined, the system may expose different ways to query quest states, one of them is to use some long living channel (let's say WebSockets).
+
+```mermaid
+sequenceDiagram
+    Explorer/Kernel ->> Quest Server: Send subscription request
+    loop
+        Quest Server -->> Explorer/Kernel: Send quest updates
+    end
+```
+
+### Workflows: Quest Designer 
+
+- Deploy new quest or modify existing quest
+
+When the quest definition is ready, the Quest Designer tool would be able to send the new or modified version to the Quest Server. The request should include the author signature.
```

Comment:
We could add something here about how quests being updated would impact players already tracking progress. We might be able to defer this decision and assume progress would be lost and invalidated.